IPython/core/tests/test_interactiveshell.py

```diff
@@ -922,3 +922,14 @@ def test_no_dep(self):
         with tt.AssertNotPrints("I AM  A WARNING"):
             ip.run_cell("wrn()")
         ip.run_cell("del wrn")
+
+
+def test_custom_exc_count():
```

Comment:
Just verifying my understanding of the test:
- set a custom exception handler for syntax errors
- grab cell execution count before running the cell with a syntax error
- run a cell with a syntax error
- restore default exception handler
- verify the hook to the handler was called only once
- verify the execution count was incremented by one